nemo/collections/asr/parts/submodules/conformer_modules.py

```diff
@@ -160,24 +158,31 @@ def __init__(self, d_model, kernel_size, norm_type='batch_norm'):
             in_channels=d_model, out_channels=d_model, kernel_size=1, stride=1, padding=0, bias=True
         )
 
-    def forward(self, x, pad_mask=None):
-        x = x.transpose(1, 2)
-        x = self.pointwise_conv1(x)
-        x = nn.functional.glu(x, dim=1)
-
-        if pad_mask is not None:
-            x.masked_fill_(pad_mask.unsqueeze(1), 0.0)
-
+    def do_conv_norm(self, x):
         x = self.depthwise_conv(x)
-
         if isinstance(self.batch_norm, nn.LayerNorm):
             x = x.transpose(1, 2)
             x = self.batch_norm(x)
             x = x.transpose(1, 2)
         else:
             x = self.batch_norm(x)
+        return x
 
+    def forward(self, x, pad_mask=None):
+        x = x.transpose(1, 2)
+        x = self.pointwise_conv1(x)
+        x = nn.functional.glu(x, dim=1)
+
+        if pad_mask is not None:
+            x.masked_fill_(pad_mask.unsqueeze(1), 0.0)
+
+            if torch.onnx.is_in_onnx_export():
+                with torch.cuda.amp.autocast(enabled=False):
```

Comment:
Is this part necessary as we already have these autocasts and converisons already in the BatchNorm1dNoAutoCast?
BatchNorm1dNoAutoCast just misses the last .to(dtype=x.dtype) which can be added.

Reply:
We are not using BatchNorm1dNoAutoCast here - I had to use that helper class in Citrinet because there batch conv's forward() was not called explicitly. Here it is explicit, and I wanted to keep it fused with previous conv, so I did not use helper class.

Reply:
(changed to BatchNorm1dNoAutoCast for clarity)